modules/application-health-about.adoc

```diff
@@ -9,121 +9,165 @@ A probe is a Kubernetes action that periodically performs diagnostics on a
-
-For example, a liveness probe on a node with a `restartPolicy` of `Always` or `OnFailure`
++
+For example, a Liveness check on a node with a `restartPolicy` of `Always` or `OnFailure`
 kills and restarts the Container on the node.
 
-.Sample Liveness Check
+Startup check::
+Legacy applications can require additional startup time on their first initialization. This situation can make it difficult to set up liveness probe parameters without compromising the fast response to deadlocks that a liveness probe provides. To prevent containers from starting slowly, configure a startup probe using the same command, HTTP or TCP check, with `failureThreshold` and `periodSeconds` values long enough to handle the worst case startup time.
++
+For example, you can add a Startup check with a maximum of 5 minutes (30 * 10 = 300s) to a Liveness check. After the Startup check succeeds the first time, the liveness probe takes over to provide a fast response to container deadlocks. If the startup probe never succeeds within the specified time period, the container is killed and is subject to the pod `restartPolicy`.
+
+Types of probes::
+Liveness, Readiness, and Startup checks can be configured in three ways:
++
+--
+* HTTP probe. When using an HTTP probe, the kubelet determines the healthiness of the container by using a web hook. This probe is successful if the HTTP response code is between 200 and 399.
++
+You can use an HTTP probe with applications that return HTTP status codes when completely initialized.
+
+* Container Execution probe. When using a Container Execution probe, the kubelet executes a command inside the container. The probe is successful if the probe exits with
+a `0` status.
+
+* TCP Socket probe. When using a TCP Socket probe, the kubelet attempts to open a socket to the container. The container is only
+considered healthy if the probe can establish a connection. You can use a TCP socket probe with applications that do not start listening until
+initialization is complete.
+--
++
+--
+Probes have a number of fields that you can use to more precisely control the behavior of Liveness and Readiness checks:
+
+* `initialDelaySeconds`: The time in seconds after the container has started before the probe can be scheduled. The defaults is `0`.
+* `periodSeconds`: The time in seconds that after which the probe is performed. The default is `10`.
+* `timeoutSeconds`: The time in seconds after which the probe times out. The defaults is `1`.
+* `successThreshold`: The minimum number of consecutive successes for the probe to be considered successful after a failure. The value must be `1` for a Liveness check. The default is `1`. 
+* `failureThreshold`: The number of times to retry the probe after a failure. After the specified retries for a Liveness check, the container is restarted. For a Readiness check, the pod is marked `Unready`. For a Startup check, the container is killed and is subject to the pod's `restartPolicy`. The defaults is 3.
```

Comment:
I am not sure if the definition of "failureThreshold" is correct, I think it should be the number of times that the probe is allowed to fail before the liveness probe restarts the container (or in the case of a readiness probe marks the pod as unavailable). I got it from https://www.openshift.com/blog/liveness-and-readiness-probes.
for example ,assuming failureThreshold = 30, according to you, the proble is allowed to fail 31 times, yet according to me, it is 30 times.